### PR TITLE
feat: choose the number of output vtu files

### PR DESCRIPTION
**Pull request by untereiner**

### CODE REVIEW

**Comment by Algiane** [comment]
I am interested by this feature too: thanks for the PR.

**Comment by untereiner** [comment]
It works as it is but you should be aware that their is no correlation for time-dependent outputs. It means that:
- it is correct to see a property for each time step 
- computing something between two timesteps is not ok because the merging can change the order of the elements.

It should be possible to keep an _absolute order_ by outputting the `localToGlobalMap` attribute

**Comment by untereiner** [comment]
To be more precise. It is not possible to keep the coherence of rank merging between each time step. This means that a cell can be in different multi blocks at each time step. 

How to keep the consistency between the time steps, meaning: Get a properly from the same cell for each time step ?
I propose to use `localToGlobalMap` to reference the cells (and access to properties) because it is stable and across time.

**Comment by herve-gross** [comment]
@untereiner is this ready for reviews? Can you explain why indices are shuffled across time steps?

**Comment by untereiner** [comment]
@herve-gross yes it is ready. The indices are not shuffled.
Currently there is an isomorphism between the ranks in memory and the ranks written in a multiblock.
There is not possible to enforce which ranks that will be merged. Then for each time step you will have different ranks that will be merged into blocks.
So, if you go through the timesteps with paraview the resulting color map will be ok because it loads and renders each timestep independently.
But we observed that computing a difference between two timesteps will result in incoherent results because filters are applied per block.

**Comment by paveltomin** [comment]
Waiting on code owner review from @MelReyCG, @rrsettgast, and/or @wrtobin.

File: src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp
Comment (rrsettgast) [inline src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp:1194]: Is this to avoid writing empty regions to ranks that don't contain any elements in them?

**Review by rrsettgast — APPROVED** [review]

File: src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp
Comment (untereiner) [reply src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp:1194]: @rrsettgast yes it is

**Comment by paveltomin** [comment]
requires rebaseline because of newly added `numberOfTargetProcesses`

File: src/coreComponents/fileIO/Outputs/VTKOutput.cpp
Comment (MelReyCG) [inline src/coreComponents/fileIO/Outputs/VTKOutput.cpp]: I would expect something like:
```suggestion
    setDescription( "Number of output aggregate files to be written." );
```

File: src/coreComponents/fileIO/Outputs/VTKOutput.cpp
Comment (MelReyCG) [inline src/coreComponents/fileIO/Outputs/VTKOutput.cpp:116]: Valid values seem to be between 1 & MPI::commSize(), what do you think of adding `GEOS_ERROR()`s or at least `GEOS_WARNING()`s.
Source: https://vtk.org/doc/nightly/html/classvtkAggregateDataSetFilter.html#:~:text=void%20vtkAggregateDataSetFilter%3A%3ASetNumberOfTargetProcesses

File: src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.hpp
Comment (MelReyCG) [inline src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.hpp]: I may I missed something, but does anything inherits from `VTKPolyDataWriterInterface`?

File: src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp
Comment (MelReyCG) [inline src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp:1321]: I would add a small comment for that:
```suggestion
    // any rank that does not hold data do not participate in the output
    globalValues.erase( std::remove_if( globalValues.begin(),
```

**Review by MelReyCG — COMMENTED** [review]
minor comments

File: src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.hpp
Comment (untereiner) [reply src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.hpp]: can be removed here because it is kept their: https://github.com/GEOS-DEV/GEOS/pull/2957/files#r1742425368

**Pull request merged**
